src/Rules/Generics/GenericObjectTypeCheck.php

```diff
@@ -84,8 +89,35 @@ public function check(
 				}
 
 				$templateType = $templateTypes[$i];
-				$boundType = TemplateTypeHelper::resolveToBounds($templateType);
 				$genericTypeType = $genericTypeTypes[$i];
+
+				$genericTypeVariance = $genericTypeVariances[$i] ?? TemplateTypeVariance::createInvariant();
+				if ($templateType instanceof TemplateType && !$genericTypeVariance->invariant()) {
+					if ($genericTypeVariance->equals($templateType->getVariance())) {
+						$messages[] = RuleErrorBuilder::message(sprintf(
+							$typeProjectionIsRedundantMessage,
```

Comment:
This is not technically a problem. I feel like it should be just a warning, but afaik PHPStan doesn't know that concept, so I've included it here along with a tip on how to fix it